backend/public/functions.js

```diff
@@ -0,0 +1,563 @@
+        ["group", "name", "hardware"].includes(column),
+        `Invalid column ${column}`
+      );
+      console.assert(column in b.record, `Missing column ${column} in b ${b}`);
+
+      // These are just names so we want to sort them with the lower value first
+      return a.record[column] > b.record[column];
+    }
+    else {
+      console.assert(
+        ["pixelError", "timing", "commitHash", "timeStamp"].includes(column),
+        `Invalid column ${column}`
+      );
+      console.assert(a.record.data.length > 0, `No test records in a ${a}`);
+      console.assert(b.record.data.length > 0, `No test records in b ${b}`);
+
+      // These are values that exist in the individual records and we want to sort based
+      // on the most recent value, which is at the front of the list
+      let aData = a.record.data[0];
+      let bData = b.record.data[0];
+      console.assert(column in aData, `Missing column ${column} in a ${a}`);
+      console.assert(column in bData, `Missing column ${column} in b ${b}`);
+
+      let aVal = aData[column];
+      let bVal = bData[column];
+
+      if (column === "timeStamp") {
+        // The time stamp gets passed to us as an ISO string, which we need to convert
+        // into a Date object to do a proper comparison
+        return new Date(aVal) > new Date(bVal);
+      }
+      else if (column === "pixelError") {
+        // The pixel error is the only value that we want to sort inverted, meaning that
+        // we want the row with the biggest error at the top
+        return aVal < bVal;
+      }
+      else {
+        return aVal > bVal;
+      }
+    }
+  }
+
+  lis.sort(sortFunc);
+  let newList = list.cloneNode(false);
+  // Add the header at the top again
+  newList.appendChild(list.childNodes[0]);
+  for (let li of lis) {
+    newList.appendChild(li);
+  }
+  list.parentNode.replaceChild(newList, list);
+} // function sortRows(column)
+
+
+async function updateReferenceImage(record) {
+  let authentication = document.getElementById("admin").value;
+
+  let response = await fetch(
+    "/api/update-reference",
+    {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify({
+        adminToken: authentication,
+        group: record.group,
+        name: record.name,
+        hardware: record.hardware
+      })
+    },
+  );
+
+  if (response.status === 200) {
+    alert("Successfully updated reference image")
+  }
+  else {
+    alert(`Error updating reference\n${response.status}\nError: ${response.statusText}`);
+  }
+} // async function updateReferenceImage(record)
+
+
+function createHeader(ul) {
+  let li = document.createElement("li");
+  li.className = "heading";
+  ul.appendChild(li);
+
+  let status = document.createElement("div");
+  status.className = "cell status";
+  status.onclick = () => sortRows("pixelError");
+  status.appendChild(document.createTextNode("Error"));
+  li.appendChild(status);
```

Comment:
In my view it would be less readable since we have to pass a lot of the interesting bits into the function anyway, would be:
```
function createDiv(className, sortColumn, name) {
  const div= document.createElement("div");
  div.className = `cell ${className}`;
  div.onclick = () => sortRows(sortColumn);
  div.appendChild(document.createTextNode(name));
  return div;
};

li.appendChild(createDiv("status", "pixelError", "Error"));
li.appendChild(createDiv("group", "group", "Group"));
li.appendChild(createDiv("name", "name", "Name"));
li.appendChild(createDiv("hardware", "hardware", "Hardware"));
```

Which makes it harder for me to read the code as I now have to understand what the `createDiv` function does?